api/src/internal/oscc.h

```diff
@@ -58,5 +74,46 @@ oscc_result_t oscc_disable_throttle(
 
 void oscc_update_status( );
 
+oscc_result_t register_can_signal();
+
+// Enables asynchronous callback to each socket should be started after all
+// connections are made
+oscc_result_t oscc_async_enable(
+    int socket );
+
+// Runs a calback function through all available socketcan signals
+oscc_result_t oscc_search_can(
+    struct can_contains(*search_callback)( const char * ),
```

Comment:
Maybe I'm behind on this code base's convention but I'm not used to seeing the `struct` keyword. If the structs are `typedef`'d you can used the name with the `_s` you're defining and can eliminate the `struct` keyword.

Reply:
Same holds for the code below.